src/ape_ethereum/ecosystem.py

```diff
@@ -216,19 +216,20 @@ def str_to_slot(text):
 
             return ProxyInfo(type=type, target=target)
 
-        # gnosis safe stores implementation in slot 0, read `masterCopy()` to be sure
+        # gnosis safe stores implementation in slot 0, read `NAME()` to be sure
         abi = MethodABI(
             type="function",
-            name="masterCopy",
+            name="NAME",
             stateMutability="view",
-            outputs=[ABIType(type="address")],
+            outputs=[ABIType(type="string")],
         )
         try:
-            master_copy = ContractCall(abi, address)()
+            name = ContractCall(abi, address)()
             storage = self.provider.get_storage_at(address, 0)
-            slot_0 = self.conversion_manager.convert(storage[-20:].hex(), AddressType)
-            if master_copy == slot_0:
-                return ProxyInfo(type=ProxyType.GnosisSafe, target=master_copy)
+            target = self.conversion_manager.convert(storage[-20:].hex(), AddressType)
```

Comment:
Actually, this is good input sanitation, if it returns an empty value we probably made a mistake somewhere. I might even raise an error if that's the case because it should never happen

Reply:
I never did this until [f3e190f](https://github.com/ApeWorX/ape/pull/1178/commits/f3e190fd45f1e76b44400865c7f5ef664ed7a390), but my initial intuiution is wrong. This value will be `ZERO_ADDRESS` on the master copy of the contract, but `NAME` will be set because it is a constant value, so it is better to check that the `target` is non-zero if the `NAME` condition holds